메인 페이지 내 북마크 리스트 섹션 및 북마크 리스트 페이지 구현

hjthebunny opened the pull request:
1. `app.js`
- 상품 데이터, 북마크 데이터 상태 관리 
- bookmarkHandler함수 선언
- 하위 컴포넌트들에 props로 전달

2. `src/pages/Main`
- 전달 받은 북마크 데이터를 불러와 렌더링
- 북마크 데이터 정렬 문제 -> margin 조정
- 북마크 리스트 섹션에 height를 주어 북마크 데이터가 없을 때에도 일정한 공간을 차지할 수 있도록 했습니다.

3. `src/components/Bookmark`
- props로 가져온 BookmarkData를 data 변수에 할당하여 상품리스트 페이지와 동일한 코드로 페이지를 구현했습니다.

4. 기타
- CSS, 폴더 및 파일 구조 수정•정리 예정입니다.
- README.md 추가될 예정입니다.

hjthebunny committed:
메인 페이지 북마크 리스트 섹션 및 북마크 리스트 페이지 구현

hjthebunny committed:
데이터 fetch 로직 수정 및 css 수정

apeachicetea commented on src/App.js:
```diff
@@ -1,24 +1,63 @@
+  const [data, setData] = useState([])
+  const [bookmarkData, setBookmarkData] = useState([])
+
+  useEffect(() => {
```
App 컴포넌트에서 전체 상품 데이터를 불러오는 것은 바람직하지 않습니다.
그 이유는 실제 데이터가 필요한 페이지는 메인페이지이기 때문입니다.
메인페이지로 옮긴 후 count 쿼리를 사용하여 필요한 양만 받아오게끔 작성해주세요!

apeachicetea commented on src/components/Product/Product.js:
```diff
@@ -1,70 +1,101 @@
+    CATEGORY: 'Category',
+    EXHIBITION: 'Exhibition',
+    PRODUCT: 'Product',
+    BRAND: 'Brand'
```
object.freeze를 활용하여 나도 모르게 수정될 가능성을 방지할 수 있도록 해주세요!

apeachicetea commented on src/pages/Bookmark/Bookmark.js:
```diff
@@ -0,0 +1,52 @@
+import React, { useState } from "react";
+import Product from "../../components/Product/Product";
+import * as S from "./style";
+const Type = {
```
해당 타입을 담은 객체가 중복되서 사용되고 있습니다.
타입을 별도로 파일링해서 export 한 후 필요한 곳에 import해서 사용할 수 있도록 해주세요!

apeachicetea commented:
전반적으로 잘해주셨습니다! 
말씀하신 CSS, 폴더 및 파일 구조 수정•정리와 README.md 추가만 신경쓰셔서 마무리 잘해주세요!
고생많으셨습니다! 👍

hjthebunny committed:
피드백 반영 수정

- Type enum 객체를 개별 파일에 넣어두고 필요한 곳에 import하는 방식으로 변경
- styled components 파일명 변경
- app.js가 아닌 각 페이지에서 데이터를 fetch하는 것으로 변경
